Extensions/BBText/bbtextruntimeobject-pixi-renderer.js

```diff
@@ -0,0 +1,145 @@
+/**
+ * The PIXI.js renderer for the BBCode Text runtime object.
+ *
+ * @class BBTextRuntimeObjectPixiRenderer
+ * @constructor
+ * @param {gdjs.BBTextRuntimeObject} runtimeObject The object to render
+ * @param {gdjs.RuntimeScene} runtimeScene The gdjs.RuntimeScene in which the object is
+ */
+gdjs.BBTextRuntimeObjectPixiRenderer = function(runtimeObject, runtimeScene) {
+  this._object = runtimeObject;
+
+  // Load (or reset) the text
+  if (this._pixiObject === undefined) {
+    this._pixiObject = new MultiStyleText(runtimeObject._text, {
+      default: {
+        fontFamily: runtimeObject._family,
+        fontSize: `${runtimeObject._size}px`,
+        fill: runtimeObject._color,
+        tagStyle: ['[', ']'],
+        wordWrap: true,
+        wordWrapWidth: runtimeObject._wrappingWidth,
+        align: runtimeObject._align,
+      },
+    });
+
+    this._object.hidden = !runtimeObject._visible;
+  } else {
+    console.log('runtime obj', runtimeObject._wrappingWidth, this._pixiObject);
+    // this._pixiObject._texture.baseTexture.source.currentTime = 0;
+  }
+
+  runtimeScene
+    .getLayer('')
+    .getRenderer()
+    .addRendererObject(this._pixiObject, runtimeObject.getZOrder());
+
+  // Set the anchor in the center, so that the object rotates around
+  // its center
+  this._pixiObject.anchor.x = 0.5;
+  this._pixiObject.anchor.y = 0.5;
+
+  this.updatePosition();
+  this.updateAngle();
+  this.updateOpacity();
+  this.updateVisible();
+};
+
+gdjs.BBTextRuntimeObjectRenderer = gdjs.BBTextRuntimeObjectPixiRenderer;
+
+gdjs.BBTextRuntimeObjectPixiRenderer.prototype.getRendererObject = function() {
+  return this._pixiObject;
+};
+
+gdjs.BBTextRuntimeObjectPixiRenderer.prototype.ensureUpToDate = function() {
+  if (
+    !this._textureWasValid &&
+    this._pixiObject.texture &&
+    this._pixiObject.texture.valid
+  ) {
+    if (this._pixiObject._style.wordWrapWidth !== this._object._wrappingWidth) {
+      this._pixiObject._style.wordWrapWidth = this._object._wrappingWidth;
+      this._pixiObject.dirty = true;
+    }
+    if (this._object._align !== this._pixiObject._style.align) {
+      this._pixiObject._style.align = this._object._align;
+      this._pixiObject.dirty = true;
+    }
+    this.updatePosition();
+    this._textureWasValid = true;
+  }
+};
+
+/**
+ * Set the rendered width
+ * @param {number} width The new width, in pixels.
+ */
+gdjs.BBTextRuntimeObjectPixiRenderer.prototype.setWidth = function(width) {
+  this._object._wrappingWidth = width;
```

Comment:
Don't update the object from the renderer.
This is going against the traditional "flow" of data and make the object hard to reason about:
- I look at the object and see setWidth and setWrappingWidth. "Ok, seems that these two things are separate operations" I say to myself.
- Then I look at setWidth in the object, see that it's calling setWidth in the renderer, and then see that it's... updating `this._object._wrappingWidth`!

Accessing properties starting with a `_` is normally forbidden anywhere except for `this`, because these properties are "private". Renderers can *read* into their object private properties... but should not modify them. It won't crash but that means that the object behavior will depend on the renderer - so that a game running in Pixi might change behavior if someone write another renderer (in the future, for another platform or whatever).

In short: remove this line and move it into the object (actually it's already done in the object "setWrappingWidth" method). You can rename this method updateWrappingWidth.

Reply:
actually it was an error
wrapping width and width are the same. Height has no effect.
Pushed an update to remove setWrappingWidth :)